src/pages/audio-rewards-page/store/sagas.ts

```diff
@@ -284,6 +285,23 @@ export function* watchFetchUserChallenges() {
   })
 }
 
+export function* updateOptimisticListenStreak() {
+  const listenStreakChallenge: ReturnType<typeof getUserChallenge> = yield select(
+    getUserChallenge,
+    {
+      challengeId: 'listen-streak'
+    }
+  )
+  if (listenStreakChallenge?.current_step_count === 0) {
```

Comment:
Do we also need a case to remove the override if the challenge reports a step count > 0? Like if a user never closes a session, this will be stuck at 1?